app/services/nsm/assessment_syncer.rb

```diff
@@ -12,20 +12,79 @@ def initialize(claim, record:)
     end
 
     def call
-      sync_overall_comment
+      case claim.status
+      when 'part_grant'
+        sync_letter_adjustments
+        sync_call_adjustments
+        sync_overall_comment
+        sync_work_items
+        sync_disbursements
```

Comment:
given there is a lot of updates in this would be good to see it being done in a transaction

Reply:
done this, not sure how transaction works under the hood when it comes to updating associated records so I've ensured the transaction is happening relative to the claim by selecting work items and disbursements via the claim.